Library/Homebrew/cmd/uses.rb

```diff
@@ -77,24 +82,37 @@ def uses
 
   def intersection_of_dependents(use_runtime_dependents, used_formulae, args:)
     recursive = args.recursive?
+    only = if args.formula?
+      :formula
```

Comment:
I actually think this might be more readable just using `args.formula?` etc. everywhere.

Reply:
I don't think it's more readable when every other command uses `only` already for this.

Reply:
Hate to be picky about this but there are several commands that don't use `only` (`deps`, `list`, `outdated`, `create`).

The main difference that I can see is that these commands don't call external methods that take an `only` parameter. I agree, if we're going to be passing the value of `only` with e.g. `args.named.to_formulae_and_casks(only: only)` it makes sense to be `only`.

However, the distinction (especially with `uses`) is that `--formula` and `--cask` mean that we are expecting _outputs_ of formulae or casks rather than _inputs_ of formulae or casks. All of the logic is handled within `uses.rb`, so I see no reason why we should hold ourselves to a convention being used in _most_ command with a totally different use case.

To me, reading `if show_formulae_and_casks || args.formula?` is much easier to understand than `if only != :cask`. There are too many negatives in the second one. It's like saying "are we not excluding" instead of "are we including"

Reply:
> Hate to be picky about this but there are several commands that don't use `only` (`deps`, `list`, `outdated`, `create`).
> 
> The main difference that I can see is that these commands don't call external methods that take an `only` parameter. I agree, if we're going to be passing the value of `only` with e.g. `args.named.to_formulae_and_casks(only: only)` it makes sense to be `only`.

I feel like there's definitely a shared abstraction that all these commands could use to reduce the boilerplate.

Reply:
Sorry, are you talking about the commands that _do_ use `only` or the commands that _don't_. If the former, I think that already exists for the most part. We could add an option to `args` that figures out the value of `only`. It would probably save two lines of code in each file but would reduce duplication. If the latter, I think the reason that they don't use `only` is that their functions are different enough that it doesn't apply (so there likely isn't much logic to be pulled out)

Reply:
> Sorry, are you talking about the commands that _do_ use `only` or the commands that _don't_. If the former, I think that already exists for the most part.

Either! 😁 

> We could add an option to `args` that figures out the value of `only`. It would probably save two lines of code in each file but would reduce duplication.

This sounds good 👍🏻

Reply:
Okay, I'll look into it. I'm going to merge this, though, because those changes are not likely to affect the `uses` command.

(Of course, if they do, this can be modified then)

Reply:
@Rylan12 Agreed, this was definitely non-blocking feedback 👍🏻